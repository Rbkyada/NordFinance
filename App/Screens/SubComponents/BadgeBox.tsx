import React, { memo } from 'react';
import {
  View,
  StyleSheet,
  Image,
  TextStyle,
  StyleProp,
  ViewStyle,
  ImageStyle,
} from 'react-native';
import LightTheme from '@Theme/LightTheme';
import { CustomText } from '@CommonComponent/index';
import { getSize } from '@Utils/Helper';

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    borderRadius: 12,
    alignSelf: 'flex-start',
    paddingHorizontal: 15,
    paddingVertical: 4,
    overflow: 'hidden',
    gap: 8,
  },
});

interface BadgeBoxProps {
  title: string;
  isSelect?: boolean;
  isImg?: any;
  textStyle?: StyleProp<TextStyle>;
  containerStyle?: StyleProp<ViewStyle>;
  rightImg?: any;
  rightImgStyle?: StyleProp<ImageStyle>;
}

const BadgeBox = memo((props: BadgeBoxProps) => {
  const { container } = styles;
  const {
    title,
    isSelect,
    isImg,
    textStyle,
    containerStyle,
    rightImg,
    rightImgStyle,
  } = props;

  return (
    <View
      style={[
        container,
        {
          backgroundColor:
            (isSelect && LightTheme.secondary) || LightTheme.border,
        },
        containerStyle && containerStyle,
      ]}>
      {rightImg && (
        <Image
          source={{ uri: rightImg }}
          style={[getSize(10), rightImgStyle]}
        />
      )}
      <CustomText
        small
        style={[
          { color: (isSelect && LightTheme.white) || LightTheme.darkText },
          textStyle && textStyle,
        ]}>
        {title}
      </CustomText>
      {isImg && (
        <Image
          resizeMode="contain"
          source={{ uri: isImg }}
          style={{ ...getSize(10) }}
        />
      )}
    </View>
  );
});

export { BadgeBox };
