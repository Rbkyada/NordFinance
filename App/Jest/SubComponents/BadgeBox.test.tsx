import renderer from 'react-test-renderer';
import { View, Image } from 'react-native';
import { BadgeBox } from '@SubComponents/BadgeBox';

jest.mock('react-native-config', () => () => ({ run: jest.fn() }));

describe('BadgeBox', () => {
  it('renders correctly with default props', () => {
    const tree = renderer.create(<BadgeBox title="Badge" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders with custom container style', () => {
    const customStyle = { width: 100, height: 50 };
    const component = renderer.create(
      <BadgeBox title="Custom Style" containerStyle={customStyle} />,
    );
    const view = component.root.findByType(View);
    expect(view.props.style).toContainEqual(customStyle);
  });

  it('renders with right image when rightImg prop is provided', () => {
    const rightImg = 'https://example.com/right-image.png';
    const component = renderer.create(
      <BadgeBox title="Right Image" rightImg={rightImg} />,
    );
    const image = component.root.findByType(Image);
    expect(image.props.source.uri).toEqual(rightImg);
  });

  it('renders with image when isImg prop is provided', () => {
    const isImg = 'https://example.com/image.png';
    const component = renderer.create(
      <BadgeBox title="Image Badge" isImg={isImg} />,
    );
    const image = component.root.findByType(Image);
    expect(image.props.source.uri).toEqual(isImg);
  });
});
