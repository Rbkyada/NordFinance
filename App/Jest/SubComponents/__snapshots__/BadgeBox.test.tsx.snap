// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BadgeBox renders correctly with default props 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "alignSelf": "flex-start",
        "borderRadius": 12,
        "flexDirection": "row",
        "gap": 8,
        "justifyContent": "center",
        "overflow": "hidden",
        "paddingHorizontal": 15,
        "paddingVertical": 4,
      },
      {
        "backgroundColor": "rgba(243, 243, 243, 1)",
      },
      undefined,
    ]
  }
>
  <Text
    numberOfLines={0}
    small={true}
    style={
      [
        {
          "fontSize": 12,
        },
        {
          "color": "#1c1c1e",
        },
        {
          "fontFamily": "DMSans-Regular",
        },
        [
          {
            "color": "rgba(37, 42, 72, 0.2)",
          },
          undefined,
        ],
      ]
    }
  >
    Badge
  </Text>
</View>
`;
